ui/pages/confirm-transaction/confirm-transaction.component.js

```diff
@@ -19,188 +27,186 @@ import {
   ENCRYPTION_PUBLIC_KEY_REQUEST_PATH,
   DEFAULT_ROUTE,
 } from '../../helpers/constants/routes';
+import { isTokenMethodAction } from '../../helpers/utils/transactions.util';
+import { usePrevious } from '../../hooks/usePrevious';
+import {
+  getUnapprovedTransactions,
+  unconfirmedTransactionsListSelector,
+} from '../../selectors';
 import {
   disconnectGasFeeEstimatePoller,
+  getContractMethodData,
   getGasFeeEstimatesAndStartPolling,
   addPollingTokenToAppState,
   removePollingTokenFromAppState,
+  setDefaultHomeActiveTabName,
 } from '../../store/actions';
 import ConfirmTokenTransactionSwitch from './confirm-token-transaction-switch';
 import ConfTx from './conf-tx';
 
-export default class ConfirmTransaction extends Component {
-  static contextTypes = {
-    metricsEvent: PropTypes.func,
-  };
-
-  static propTypes = {
-    history: PropTypes.object.isRequired,
-    totalUnapprovedCount: PropTypes.number.isRequired,
-    sendTo: PropTypes.string,
-    setTransactionToConfirm: PropTypes.func,
-    clearConfirmTransaction: PropTypes.func,
-    mostRecentOverviewPage: PropTypes.string.isRequired,
-    transaction: PropTypes.object,
-    getContractMethodData: PropTypes.func,
-    transactionId: PropTypes.string,
-    paramsTransactionId: PropTypes.string,
-    isTokenMethodAction: PropTypes.bool,
-    setDefaultHomeActiveTabName: PropTypes.func,
-  };
-
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  _beforeUnload = () => {
-    this._isMounted = false;
-    if (this.state.pollingToken) {
-      disconnectGasFeeEstimatePoller(this.state.pollingToken);
-      removePollingTokenFromAppState(this.state.pollingToken);
+const ConfirmTransaction = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const { id: paramsTransactionId } = useParams();
+
+  const [isMounted, setIsMounted] = useState(false);
```

Comment:
Hey, useEffect is called after component rendering, not sure why we need above variable in state ?

Reply:
Hey @jpuri , thanks for bringing this up! 

I was hoping to preserve logic in this PR. I'm thinking we can chat about it here and then update in the follow-up clean-up PR if we want to apply changes, https://github.com/MetaMask/metamask-extension/pull/17546

---

re: `isMounted` variable

I think it prevents us from possible the `beforeunloaded` event is trigged before a `pollingToken` is set and the `getGasFeeEstimatesAndStartPolling` callback is called with a `pollingToken`, in which, `!isMount` would come in handy to turn off the pollling. 

ref: https://github.com/MetaMask/metamask-extension/pull/11746
> Explanation: This stops polling in the gasFeeController when a users closes confirmation screen/window with a pending transaction unfinished.

more discussion here: https://github.com/MetaMask/metamask-extension/pull/11511/files#r670686360